lib/galaxy/tools/__init__.py

```diff
@@ -2845,29 +2845,29 @@ def produce_outputs(self, trans, out_data, output_collections, incoming, history
         forward_o = incoming["input_forward"]
         reverse_o = incoming["input_reverse"]
 
-        forward, reverse = forward_o.copy(), reverse_o.copy()
+        forward, reverse = forward_o.copy(copy_tags=forward_o.tags), reverse_o.copy(copy_tags=reverse_o.tags)
         new_elements = {}
         new_elements["forward"] = forward
         new_elements["reverse"] = reverse
         self._add_datasets_to_history(history, [forward, reverse])
         output_collections.create_collection(
-            next(iter(self.outputs.values())), "output", elements=new_elements
+            next(iter(self.outputs.values())), "output", elements=new_elements, propagate_hda_tags=False
```

Comment:
In summary:
 - Adding `copy_tags` to the `produce_outputs` methods of the `DatabaseOperationTool`s ensures the dataset element tags get copied to the dataset elements in the new collections.
 - Adding `propagate_hda_tags=False` when creating the new collection ensures the dataset element tags *don't* get copied to the newly created parent collections themselves. These should only inherit the tags from the old parent collections.